alf/algorithms/data_transformer.py

```diff
@@ -597,7 +598,10 @@ def __init__(self,
         """
         super().__init__(observation_spec)
         if normalizer is None:
-            normalizer = ScalarAdaptiveNormalizer(auto_update=False)
+            normalizer = AdaptiveNormalizer(
+                tensor_spec=alf.get_reward_spec(),
```

Comment:
It doesn't feel right to couple data transformer with config util. A hypothetical scenario where this may fail is that another data transformer make change to reward spec.

Perhaps we should make the data transformer take time_step_spec as input to the constructor so that it does not need to call alf.get_reward_spec().  This can be a TODO item.

Reply:
> It doesn't feel right to couple data transformer with config util. A hypothetical scenario where this may fail is that another data transformer make change to reward spec.
> 
> Perhaps we should make the data transformer take time_step_spec as input to the constructor so that it does not need to call alf.get_reward_spec(). This can be a TODO item.

added TODO